test/counter_server_example/test/counter_server_example.erl

```diff
@@ -0,0 +1,342 @@
+%%% @doc
+%%%
+%%% Example for using vnet to simulate a cluster of nodes, one of
+%%% which serves as a counter server.
+%%%
+%%% All _test functions demonstrate scenarios of using the server. See
+%%% their documentation for more info.
+%%%
+%%% All _no_vnet_test functions do not use vnet and their purpose is
+%%% to check the core implementation of the counter server.
+%%%
+%%% @end
+%%%-------------------------------------------------------------------
+
+-module(counter_server_example).
+
+%%%-------------------------------------------------------------------
+
+-include_lib("eunit/include/eunit.hrl").
+
+-concuerror_options(
+   [ {instant_delivery, false}
+   , assertions_only
+   , {ignore_error, deadlock}
+   ]).
+
+%%%-------------------------------------------------------------------
+
+-export(
+   [ setup_server/0
+   , once_valid_client/1
+   , twice_valid_client/1
+   , invalid_client/1
+   , twice_valid_same_gen_client/1
+   ]).
+
+%%%-------------------------------------------------------------------
+
+-include("../src/counter_server.hrl").
+
+%%%-------------------------------------------------------------------
+
+%% @doc
+%% Simulating a scenario with one client, making requests and checking
+%% results, without using any vnet logic.
+valid_no_vnet_test() ->
```

Comment:
Please don't make these functions look like EUnit tests! If you run `make`, `rebar3` will try to execute them as such and fail (`{badmatch,{error,{already_started,<0.128.0>}}}`).

Reply:
ok!